test/httpProblemDetail.test.ts

```diff
@@ -0,0 +1,103 @@
+import { HttpProblemDetail } from '../src';
+import Joi from '@hapi/joi';
+
+export const problemSchema = Joi.object({
+  type: Joi.string(),
+  title: Joi.string(),
+  status: Joi.number().required(),
+}).required();
+
+describe('given HttpProblemDetail', () => {
+  describe('when adding additional details', () => {
+    let problem: HttpProblemDetail;
+    beforeEach(() => {
+      problem = new HttpProblemDetail({
+        status: 403,
+        title: 'You do not have enough credit.',
+        type: 'https://example.com/probs/out-of-credit',
+      });
+    });
+
+    it('throw an error when try to overrite status', () => {
+      expect(() => problem.addAdditionalDetail('status', 250)).toThrowError();
+    });
+
+    it('throw an error when try to overrite type', () => {
+      expect(() =>
+        problem.addAdditionalDetail(
+          'type',
+          'https://example.com/probs/not-enough-minerals'
+        )
+      ).toThrowError();
+    });
+
+    it('throw an error when try to overrite title', () => {
+      expect(() =>
+        problem.addAdditionalDetail('title', 'Not enough minerals')
+      ).toThrowError();
+    });
+  });
+
+  describe('when parse to JSON', () => {
+    let problem: HttpProblemDetail;
+    let problemSchema: Joi.ObjectSchema<any>;
+    beforeEach(() => {
+      problem = new HttpProblemDetail({
+        status: 403,
+        title: 'You do not have enough credit.',
+        type: 'https://example.com/probs/out-of-credit',
+      });
+      problemSchema = Joi.object({
+        type: Joi.string().required(),
+        title: Joi.string().required(),
+        status: Joi.number().required(),
+      }).required();
+    });
+
+    it('return passed basic params', () => {
+      expect(() => {
+        Joi.assert(problem.toJson(), problemSchema);
+      }).not.toThrow();
+    });
+
+    it('return additional passed params', () => {
+      const detailName = 'lorem';
+      problem.addAdditionalDetail(detailName, 'ipsum');
+      expect(() => {
+        Joi.assert(
+          problem.toJson(),
+          problemSchema.concat(
+            Joi.object({
+              [detailName]: Joi.string().required(),
+            })
+          )
+        );
+      }).not.toThrow();
+    });
+
+    it('ignore empty params', () => {
+      const detailName = 'lorem';
+      problem.addAdditionalDetail(detailName, '');
```

Comment:
i would move this test somewhere else, If we want to keep BDD naming style I would do it like:
```
describe('given HttpProblemDetail', () => {
  describe('given blank additional detail is set', () => {
    describe('given some additional detail is set', () => {
      describe('when parse to JSON', () => {
        it('have included only not blank additional details', () => {
```